Process.c

```diff
@@ -787,7 +787,7 @@ void Process_printLeftAlignedField(RichString* str, int attr, const char* conten
 }
 
 void Process_printPercentage(float val, char* buffer, size_t n, uint8_t width, int* attr) {
-   if (val >= 0) {
+   if (isNonnegative(val)) {
```

Comment:
Personal note. With this expression changed to using `isNonnegative`, there is an implicit promotion from `float` to `double`, is the compiler smart enough to realize the promotion can be optimized out and make the expression identical to `isgreaterequal((float)val, 0.0F)` ?

Reply:
Update. I figured it out by myself. The answer is yes.
`isNonnegative(val)` optimizes to `isgreaterequal((float)val, 0.0F)` with GCC and Clang with `-O1` or higher.

https://godbolt.org/z/h5G1rWf5h